src/main.cc

```diff
@@ -940,116 +892,166 @@ mainReconfigureFinish(void *)
                ") requires a full restart. It has been ignored by reconfigure.");
         Config.workers = oldWorkers;
     }
+    leave_suid();
 
     RunRegisteredHere(RegisteredRunner::syncConfig);
+    reconfiguring = 0;
+}
 
-    if (IamPrimaryProcess())
-        CpuAffinityCheck();
-    CpuAffinityReconfigure();
+void
+SignalEngine::doReconfigure()
```

Comment:
Part of this PR is trying to avoid/reduce C-style pre-declaration of static function. The mainReconfigureFinish() needs to be defined before it is used.